Remove genemu dependency

diimpp opened the pull request:
Fixes #20
I didn't run test suite as it require mongo. But change is trivial, should be fine. Please test if before merge.

mremi commented:
Hi @diimpp , thanks for this. I also think the parameter should be removed.

mremi commented:
About the unit tests, no need mongo. Did you get any error?

diimpp commented:
About tests, my bad, It's okay. I've run it in wrong manner and it started testing of vendors.

I see that people using 'disabled' for dev in #18 
And I don't see any explicit injection of "disabled" value. Where is the magic? :smile: 
https://github.com/mremi/ContactBundle/blob/master/Resources/config/form.xml#L8-L13

mremi commented:
The magic is here https://github.com/mremi/ContactBundle/blob/master/DependencyInjection/MremiContactExtension.php#L92 :)

For people who configures `disabled` by environment, they will do the same with `captcha_type`. So you have to remove `isRequired` from the configuration.

Moreover, can you create a `CHANGELOG.md` file please like:

CHANGELOG
=========

### 2015-01-05
 * [BC BREAK] Removed Genemu dependency.

diimpp commented:
I've removed disabled and added changelog. Looks like completed.
I've ended with removing cannotBeEmpty as well as isRequired. It don't distinguish between empty string and false. As a result I've removed tests for this value.

By the way, what tests for configuration.php are testing? There is already a constraints like cannotBeEmpty or isRequired.

mremi commented on DependencyInjection/Configuration.php:
```diff
@@ -45,8 +45,7 @@ public function getConfigTreeBuilder()
                         ->scalarNode('subject_provider')->defaultValue('mremi_contact.subject_provider.noop')->cannotBeEmpty()->end()
-                        ->booleanNode('captcha_disabled')->defaultFalse()->end()
-                        ->enumNode('captcha_type')->defaultValue('genemu_captcha')->values(array('genemu_captcha', 'genemu_recaptcha'))->end()
+                        ->scalarNode('captcha_type')->defaultFalse()->end()
```
I think this is better, no mix between boolean and string and force the final user to provide a value if configured:

```php
->scalarNode('captcha_type')->defaultNull()->cannotBeEmpty()->end()
```

diimpp replied:
I think it's a regular practice to have string|false. But how to disable it in dev than? Redefine with empty string?

User may not want to use captcha at all. Why to force him?

>By the way, what tests for configuration.php are testing? There is already a constraints like >cannotBeEmpty or isRequired.

I think this particular test is useless, all they testing is how already tested config component returning default values.  Is there purpose for it?

mremi commented on Form/Type/ContactType.php:
```diff
@@ -50,14 +45,12 @@ class ContactType extends AbstractType
      * @param string                   $class           The Contact class namespace
-     * @param boolean                  $captchaDisabled TRUE whether you want disable the captcha
-     * @param string                   $captchaType     The captcha type
+     * @param mixed                    $captchaType     The captcha type or FALSE whether you want disable the captcha
```
`or NULL` to be consistent with the configuration

mremi commented on README.md:
```diff
@@ -160,12 +174,11 @@ mremi_contact:
         subject_provider:  mremi_contact.subject_provider.noop
-        captcha_disabled:  false
-        captcha_type:      genemu_captcha # or genemu_recaptcha
+        captcha_type:      false
```
This should be removed like the previous config you added. I think the captcha configuration must have a dedicated documentation after this one.

diimpp replied:
You want me to remove captcha_type mentions in default configuration snippet and previous snippet entirely?

mremi replied:
Yes because the default value will be `NULL` unless you configure it, and this snippet shows the default values. This is useful for a quick start when you install the bundle.
I think a dedicated snippet after default values should be more appropriate and allows to highlight the captcha config.

mremi commented:
`isRequired` forces the user to add the node, regardless of its value.
`cannotBeEmpty` forces the user to configure the node (if provided) with a non-empty value.

diimpp commented:
I think it's a regular practice to have string|false. But how to disable it in dev than? Redefine with empty string?

User may not want to use captcha at all. Why force him?

> By the way, what tests for configuration.php are testing? There is already a constraints like cannotBeEmpty or isRequired.

I think this particular test is useless, all it testing is how already tested config component returning default values. Is there purpose for it?

mremi commented:
> But how to disable it in dev than? Redefine with empty string?

Using `config_dev.yml` and `config_prod.yml`.

> User may not want to use captcha at all. Why force him?

I agree and I don't want to force him, with `->scalarNode('captcha_type')->defaultNull()->cannotBeEmpty()->end()` the captcha is disabled by default, you have to configure the node to activate it (only a non-empty value is mandatory, not the node itself).

> I think this particular test is useless, all it testing is how already tested config component returning default values. Is there purpose for it?

The purpose is not to test the Config component, just make sure any changes on the configuration still do what I expect.

diimpp commented:
Hm, okay, I see now.  Last question with readme mentions. I've asked above.

diimpp commented:
Done.

diimpp commented:
Thank you. That was a long way. :)

mremi commented:
Yep, but got it :)
